Test failure looks related: 
```
ERROR: corehq.apps.es.tests.test_case_search_es:TestCaseSearchLookups.test_get_related_case_results
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/es/tests/test_case_search_es.py", line 507, in test_get_related_case_results
    self._assert_related_case_ids(cases, {"parent/parent"}, {"c1"})
  File "/mnt/commcare-hq-ro/corehq/apps/es/tests/test_case_search_es.py", line 513, in _assert_related_case_ids
    results = get_related_case_results(self.domain, cases, paths)
  File "/mnt/commcare-hq-ro/corehq/apps/case_search/utils.py", line 266, in get_related_case_results
    indices = [case.get_index(identifier) for case in current_cases]
  File "/mnt/commcare-hq-ro/corehq/apps/case_search/utils.py", line 266, in <listcomp>
    indices = [case.get_index(identifier) for case in current_cases]
AttributeError: 'dict' object has no attribute 'get_index'
```

--- a/corehq/apps/case_search/exceptions.py
+++ b/corehq/apps/case_search/exceptions.py
@@ -4,3 +4,7 @@ class CaseSearchException(Exception):
 
 class CaseSearchNotEnabledException(CaseSearchException):
     pass
+
+
+class CaseSearchUserError(CaseSearchException):
+    pass

--- a/corehq/apps/case_search/tests/test_case_search_endpoint.py
+++ b/corehq/apps/case_search/tests/test_case_search_endpoint.py
@@ -1,0 +1,51 @@
+import uuid
+
+from django.test import TestCase
+
+from casexml.apps.case.mock import CaseBlock
+
+from corehq.apps.case_search.models import CaseSearchConfig
+from corehq.apps.domain.shortcuts import create_user
+from corehq.apps.es.tests.utils import (
+    case_search_es_setup,
+    case_search_es_teardown,
+    es_test,
+)
+from corehq.form_processor.tests.utils import run_with_sql_backend
+
+from ..utils import get_case_search_results
+
+
+@es_test
+@run_with_sql_backend
+class TestCaseSearchEndpoint(TestCase):
+    domain = "TestCaseSearchEndpoint"
+
+    # TODO convert to setUpClass after it stabilizes
+    def setUp(self):
+        self.user = create_user("admin", "123")
+        CaseSearchConfig.objects.create(domain=self.domain, enabled=True)
+        case_search_es_setup(self.domain, [
+            CaseBlock(
+                case_id=str(uuid.uuid4()),
+                case_type='person',
+                case_name=name,
+                create=True,
+                update=properties,
+            ) for name, properties in [
+                ("Jane", {"family": "Villanueva"}),
+                ("Xiomara", {"family": "Villanueva"}),
+                ("Alba", {"family": "Villanueva"}),
+                ("Rogelio", {"family": "de la Vega"}),
+                ("Jane", {"family": "Ramos"}),
+            ]
+        ])
+
+    def tearDown(self):
+        case_search_es_teardown()
+
+    def test_no_filter(self):
+        res = get_case_search_results(self.domain, {'case_type': 'person'})
+        self.assertItemsEqual(["Jane", "Xiomara", "Alba", "Rogelio", "Jane"], [
+            case.name for case in res
+        ])

--- a/corehq/apps/case_search/utils.py
+++ b/corehq/apps/case_search/utils.py
@@ -1,9 +1,18 @@
 import re
+
 from django.utils.translation import ugettext as _
 
 from casexml.apps.case.models import CommCareCase
+from dimagi.utils.logging import notify_exception
+
 from corehq.apps.app_manager.dbaccessors import get_app_cached
 from corehq.apps.app_manager.util import module_offers_search
+from corehq.apps.case_search.const import CASE_SEARCH_MAX_RESULTS
+from corehq.apps.case_search.exceptions import CaseSearchUserError
+from corehq.apps.case_search.filter_dsl import (
+    CaseFilterError,
+    TooManyRelatedCasesError,
+)
 from corehq.apps.case_search.models import (
     CASE_SEARCH_BLACKLISTED_OWNER_ID_KEY,
     CASE_SEARCH_XPATH_QUERY_KEY,
@@ -13,8 +22,41 @@ from corehq.apps.case_search.models import (
     FuzzyProperties,
 )
 from corehq.apps.es.case_search import CaseSearchES, flatten_result
-from corehq.apps.case_search.const import CASE_SEARCH_MAX_RESULTS
-from corehq.apps.case_search.filter_dsl import CaseFilterError
+
+
+def get_case_search_results(domain, criteria, app_id=None):
+    try:
+        # could be a list or a single string
+        case_type = criteria.pop('case_type')
+        case_types = case_type if isinstance(case_type, list) else [case_type]
+    except KeyError:
+        raise CaseSearchUserError(_('Search request must specify case type'))
+
+    try:
+        case_search_criteria = CaseSearchCriteria(domain, case_types, criteria)
+    except TooManyRelatedCasesError:
+        raise CaseSearchUserError(_('Search has too many results. Please try a more specific search.'))
+    except CaseFilterError as e:
+        # This is an app building error, notify so we can track
+        notify_exception(None, str(e), details=dict(
+            exception_type=type(e),
+        ))
+        raise CaseSearchUserError(str(e))
+    search_es = case_search_criteria.search_es
+
+    try:
+        hits = search_es.run().raw_hits
+    except Exception as e:
+        notify_exception(None, str(e), details=dict(
+            exception_type=type(e),
+        ))
+        raise
+
+    # Even if it's a SQL domain, we just need to render the hits as cases, so CommCareCase.wrap will be fine
+    cases = [CommCareCase.wrap(flatten_result(result, include_score=True)) for result in hits]
+    if app_id:
+        cases.extend(get_related_cases(domain, app_id, case_types, cases))
+    return cases
 
 
 class CaseSearchCriteria(object):

--- a/corehq/apps/es/tests/utils.py
+++ b/corehq/apps/es/tests/utils.py
@@ -84,7 +84,8 @@ def populate_user_index(users):
 def case_search_es_setup(domain, case_blocks):
     """Submits caseblocks, creating the cases, then sends them to ES"""
     from corehq.apps.hqcase.utils import submit_case_blocks
-    _, cases = submit_case_blocks([cb.as_text() for cb in case_blocks], domain=domain)
+    xform, cases = submit_case_blocks([cb.as_text() for cb in case_blocks], domain=domain)
+    assert not xform.is_error, xform
     order = {cb.case_id: index for index, cb in enumerate(case_blocks)}
     # send cases to ES in the same order they were passed in
     cases = sorted(cases, key=lambda case: order[case.case_id])

--- a/corehq/apps/ota/views.py
+++ b/corehq/apps/ota/views.py
@@ -38,15 +38,14 @@ from corehq.apps.app_manager.dbaccessors import (
 )
 from corehq.apps.app_manager.models import GlobalAppConfig
 from corehq.apps.builds.utils import get_default_build_spec
-from corehq.apps.case_search.filter_dsl import CaseFilterError, TooManyRelatedCasesError
-from corehq.apps.case_search.utils import CaseSearchCriteria, get_related_cases
+from corehq.apps.case_search.exceptions import CaseSearchUserError
+from corehq.apps.case_search.utils import get_case_search_results
 from corehq.apps.domain.decorators import (
     check_domain_migration,
     mobile_auth,
     mobile_auth_or_formplayer,
 )
 from corehq.apps.domain.models import Domain
-from corehq.apps.es.case_search import flatten_result
 from corehq.apps.locations.permissions import location_safe
 from corehq.apps.ota.decorators import require_mobile_access
 from corehq.apps.ota.rate_limiter import rate_limit_restore
@@ -107,42 +106,12 @@ def app_aware_search(request, domain, app_id):
         Multiple values can be specified for a param, which will be searched with OR operator
 
     Returns results as a fixture with the same structure as a casedb instance.
-
-
     """
     criteria = {k: v[0] if len(v) == 1 else v for k, v in request.GET.lists()}
     try:
-        # could be a list or a single string
-        case_type = criteria.pop('case_type')
-        case_types = case_type if isinstance(case_type, list) else [case_type]
-    except KeyError:
-        return HttpResponse('Search request must specify case type', status=400)
-
-    try:
-        case_search_criteria = CaseSearchCriteria(domain, case_types, criteria)
-    except TooManyRelatedCasesError:
-        return HttpResponse(_('Search has too many results. Please try a more specific search.'), status=400)
-    except CaseFilterError as e:
-        # This is an app building error, notify so we can track
-        notify_exception(request, str(e), details=dict(
-            exception_type=type(e),
-        ))
+        cases = get_case_search_results(domain, criteria, app_id)
+    except CaseSearchUserError as e:
         return HttpResponse(str(e), status=400)
-    search_es = case_search_criteria.search_es
-
-    try:
-        hits = search_es.run().raw_hits
-    except Exception as e:
-        notify_exception(request, str(e), details=dict(
-            exception_type=type(e),
-        ))
-        return HttpResponse(status=500)
-
-    # Even if it's a SQL domain, we just need to render the hits as cases, so CommCareCase.wrap will be fine
-    cases = [CommCareCase.wrap(flatten_result(result, include_score=True)) for result in hits]
-    if app_id:
-        cases.extend(get_related_cases(domain, app_id, case_types, cases))
-
     fixtures = CaseDBFixture(cases).fixture
     return HttpResponse(fixtures, content_type="text/xml; charset=utf-8")
 

